list.go

```diff
@@ -0,0 +1,31 @@
+package ctxvfs
+
+import "context"
+
+// ListAllFileser is type that implements ListAllFiles
+type ListAllFileser interface {
+	// ListAllFiles returns a slice of all file paths on a VFS. This is
+	// only files (excludes directories). This interface exists for
+	// implementors to provide a faster way to list all files than walking
+	// the file tree with ReadDir/etc.
+	ListAllFiles(ctx context.Context) ([]string, error)
+}
+
+// ListAllFiles returns a slice of all file paths (excludes directories).
+func ListAllFiles(ctx context.Context, fs FileSystem) ([]string, error) {
```

Comment:
Can we make those test environments explicitly implement `ListAllFiles` with the slow path instead of an automatic fallback to the slow path if the fast path does not exist?